### PR TITLE
Ports Door Wands

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Port of: https://github.com/tgstation/-tg-station/pull/15599

Adds in door control wands.

Simple device that can toggle doors open/closed, bolt/unbol them, or grant/remove emergency access. These work from range.

Each head of staff (and the QM) gets one of these, with access to the doors in their department.

Gives head of staff greater control of their and allows for easier door opening/access granting for when there's an AI/it's low pop.

Code Related
- Changed the order of the `get_region_access`
- Ensured that science access properly included genetics

:cl: Fox McCloud
add: Adds in door wands--control doors from range
add: Each head of staff (and the QM) starts off with a door wand to control the doors in their respective departments
/:cl:

### CODE REVIEW

**Comment by monster860** [comment]
Epic. Ill use that when I need to make people go AI ROGUE

**Comment by monster860** [comment]
Also theres a typo in the cl

**Comment by FlattestGuitar** [comment]
This is actually a heavy buff against a rogue AI. Being able to unbolt doors without having the proper tools might be too much.

**Comment by QQCucumber** [comment]
That is a good point and it would be nice if the door wands weren't able to override an AI.

**Comment by Fox-McCloud** [comment]
>This is actually a heavy buff against a rogue AI. Being able to unbolt doors without having the proper tools might be too much.

If malf AI's don't disable ID scan, that's on them.

File: code/game/jobs/access.dm [hunk 1]
```diff
@@ -253,20 +253,20 @@
...
-			return list(access_heads, access_RC_announce, access_keycard_auth, access_change_ids, access_ai_upload, access_teleporter, access_eva, access_tcomsat, access_gateway, access_all_personal_lockers, access_heads_vault, access_blueshield, access_ntrep, access_hop, access_captain)
-		if(6) //station general
-			return list(access_kitchen,access_bar, access_hydroponics, access_janitor, access_chapel_office, access_crematorium, access_library, access_theatre, access_lawyer, access_magistrate, access_clown, access_mime)
-		if(7) //supply
+		if(6) //supply
 			return list(access_mailsorting, access_mining, access_mining_station, access_mineral_storeroom, access_cargo, access_qm)
+		if(7) //command
+			return list(access_heads, access_RC_announce, access_keycard_auth, access_change_ids, access_ai_upload, access_teleporter, access_eva, access_tcomsat, access_gateway, access_all_personal_lockers, access_heads_vault, access_blueshield, access_ntrep, access_hop, access_captain)
 
 /proc/get_region_accesses_name(var/code)
```
Comment (tigercat2000) [inline code/game/jobs/access.dm]: you need to update this too

File: code/game/jobs/access.dm [hunk 2]
```diff
@@ -253,20 +253,20 @@
 	switch(code)
 		if(0)
```
Comment (tigercat2000) [inline code/game/jobs/access.dm]: please replace these with `#define`d values, such as `#define REGION_ALL 0`
there is only four uses of this proc anywhere

**Comment by pinatacolada** [comment]
ID scan doesn't have a hotkey to it, it's one thing to lockdown the HoS in the armory by ctrl clicking wherever he's running in advance, it's another to click a door and then the button for it. this is a buff against the AI in general, be it being malf or just locking down a traitor head

**Comment by Spacemanspark** [comment]
AI is already extremely easy to destroy.
No.
:-1:

File: code/game/jobs/access.dm (hunk 1 above)
Comment (Fox-McCloud) [inline code/game/jobs/access.dm]: good catch.

**Comment by Regen1** [comment]
I will approve of this if a similar function to the "disable RCD's" ability is added to the malf AI

Maybe make it disable the wands whenever hostile takeover is declared

**Comment by QQCucumber** [comment]
Auto-disable wands on hostile takeover + a disable IDs **hotkey** would be good.

**Comment by Kyep** [comment]
+1 so long as they are easily disabled by AIs during malf.

File: code/game/jobs/access.dm [hunk 3]
```diff
@@ -253,20 +253,20 @@
 	switch(code)
 		if(0)
 			return get_all_accesses()
-		if(1) //security
+		if(1) //station general
+			return list(access_kitchen,access_bar, access_hydroponics, access_janitor, access_chapel_office, access_crematorium, access_library, access_theatre, access_lawyer, access_magistrate, access_clown, access_mime)
```
Comment (tigercat2000) [inline code/game/jobs/access.dm]: space after `access_kitchen,` :eyes:

File: code/game/jobs/access.dm [hunk 4]
```diff
@@ -249,43 +249,43 @@
...
 
-/proc/get_region_accesses(var/code)
+/proc/get_region_accesses(code)
 	switch(code)
-		if(0)
+		if(REGION_ALL)
 			return get_all_accesses()
-		if(1) //security
+		if(REGION_GENERAL) //station general
+			return list(access_kitchen,access_bar, access_hydroponics, access_janitor, access_chapel_office, access_crematorium, access_library, access_theatre, access_lawyer, access_magistrate, access_clown, access_mime)
```
Comment (tigercat2000) [inline code/game/jobs/access.dm]: a spacebetweenaccess_kitchen,andaccess_barmightbehelpful

**Comment by TheDZD** [comment]
If the AI can't bother to properly secure its core when it is malf, it needs to git gud.

**Comment by DorainGrey** [comment]
Hey @Fox-McCloud can I port this over to vgstation?

**Pull request merged**